python/pyspark/sql/avro/functions.py

```diff
@@ -37,7 +37,7 @@ def from_avro(data, jsonFormatSchema, options=None):
 
     Parameters
     ----------
-    data : :class:`Column` or str
+    data : :class:`pyspark.sql.Column` or str
```

Comment:
IIRC, it doesn't have to be fully qualified import path. `Column` is already being imported in this file. For example, it works for `` `:class:`DataFrame`, see https://hyukjin-spark.readthedocs.io/en/stable/reference/api/pyspark.sql.DataFrameReader.csv.html#pyspark.sql.DataFrameReader.csv and https://github.com/apache/spark/blob/master/python/pyspark/sql/readwriter.py#L527

Can you investigate why it doesn't work here with `Column`? I would like to avoid making it as a fully qualified form which seems unnecessary.

Reply:
In both cases you referenced, the docstring is part of a class in the same module as `Column`, i.e. `pyspark.sql`, which is why the name lookup works. Here, though, we're in child modules, `pyspark.sql.avro.functions` and `pyspark.sql.functions`.

I agree that having to change references to `Column` is a nuisance that I too would like to avoid, but last time I checked, I found no way to make Sphinx render as intended by setting a configuration option of some kind.

There are two variations, though, that could be used:

- One could say ``:class:`.Column` `` (note the dot before `Column`) to make Sphinx look for `Column` in all modules. This still requires changes to numerous places in the docstrings, and may cause ambiguity in case the referenced name is not unique.
- Also, one can use the `~` prefix, as in ``:class:`~pyspark.sql.Column` `` to make the rendered text say `Column`, rather than `pyspark.sql.Column`, but still link to the correct place. Which output is preferable is a matter of taste, I guess, and I'm indifferent.

If someone knows a better way to fix this, I'd be very happy to hear about it!

Reply:
Hm, but `Column` is being imported, and it will be able to be seen via `pyspark.sql.functions.Column` too:

```python
>>> from pyspark.sql.functions import Column
>>>
```

I wonder why Sphinx can't detect this automatically.

In any event, the doc structure followed pandas'. It seems like pandas uses the second way too at least for `Series`. So, doing the second way seems okay.

Reply:
I understood your comment (and a cursory galnce at the pandas docs) to mean you'd like ``:class:`~pyspark.sql.Column` ``, and have pushed a commit accordingly. If I was mistaken, please let me know.